src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -62,16 +66,11 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
                 await _lifetimeManager.OnConnectedAsync(connection);
 
                 using (var scope = _serviceScopeFactory.CreateScope())
+                using (var hubWrapper = _hubActivator.Create<THub, TClient>(scope.ServiceProvider))
```

Comment:
In MVC they either pass the serviceProvider: 
https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs#L66-L80
https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/Internal/TypeActivatorCache.cs#L23-L37
or they use requestServices from HttpContext which are request scoped:
https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BinderTypeModelBinder.cs#L51-L52